provider/azure/credentials.go

```diff
@@ -13,7 +15,11 @@ import (
 const (
 	credAttrAppId          = "application-id"
 	credAttrSubscriptionId = "subscription-id"
+	credAttrTenantId       = "tenant-id"
 	credAttrAppPassword    = "application-password"
+
+	clientCredentialsAuthType cloud.AuthType = "service-principal-secret"
+	deviceCodeAuthType        cloud.AuthType = "interactive"
```

Comment:
"device code" and "client credentials" are the OAuth terms. I gave them names that would be more meaningful to a user. Added comments to the constants.